### PR TITLE
Communicating regarding Arcade

### PR DESCRIPTION
**Pull request by markwilkie**
Here's an initial take on how we might think about communicating updates.

### CODE REVIEW

File: Documentation/ArcadeCommunications.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  - Examples: release pipelines, licensing, cert removal
+- **Minimal risk** - Some risk, but likely will be fine (these might also be called “updates of note”)
+  - Examples: new feature, notable bug fix, etc. 
+- **Minor, low risk** - Very low risk, and most folks won't care.  
+  - Most changes are these last two bullets.
+  - Examples: security update, minor bug fixes, etc
+
+## Engineering Services Policy for Updates
+- Product release schedules are published and kept "top of mind" for the team
+- Tools will **not** flow to release/shipping branches (only to master/dev)
```
Comment (mmitche) [inline Documentation/ArcadeCommunications.md]: I think we should caveat this with "automatically".  There may be the occasional case where we need to take an arcade update in a release branch (and we will surely need to do so in servicing occasionally). We should set up our system to allow for the possibility of flowing tool updates to those channels, but avoid automatic updates.

File: Documentation/ArcadeCommunications.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
+# Types of Arcade Communication
+
+## Areas to Communicate About
+There are several areas around which  communication will occur.  The nature of these updates remain the same for each of these areas.
+- DNC Services (Helix, Darc, etc) 
+- Shared Tools (Arcade SDK, Signing, etc)
```
Comment (rynowak) [inline Documentation/ArcadeCommunications.md]: What about guidance and best practices? (The recipe for doing Xyz is changing..). 

Where would you classify things like the pipelines.yml files? These aren't really *shared tools*, but they are assets that are tightly coupled to our tools and practices. They also aren't managed by arcade, each repo customized them.

File: Documentation/ArcadeCommunications.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+- Product release schedules are published and kept "top of mind" for the team
+- Tools will **not** flow to release/shipping branches (only to master/dev)
+- No updates to shared services during stabilization period (often a week) leading up to release
+- Published bar for when we’ll take changes (*link to doc here*)
+- Published bar used for determining class of update/change (*link to bar document*)
+
+## Communication Mechanics/Methods
+- Breaking changes follow procedure  (*add link here*) 
+- Disruptive changes follow procedure (*add link here*)
+- Release notes with “updates of note” (human crafted)
```
Comment (rynowak) [inline Documentation/ArcadeCommunications.md]: +1 x a million for this.

File: Documentation/ArcadeCommunications.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  - Done every time an update is deployed for our services
+  - Done at most weekly for tools (e.g. Arcade SDK).
+  - Done when items of note occur in one of our backing resources.  For example, public feeds for Azure Artifacts
+- Blog entry for larger features/updates of interest
```
Comment (rynowak) [inline Documentation/ArcadeCommunications.md]: Does it also fit into this category to share some reporting of overall status and health? 
- Propagation time of dependencies?
- Number of E2E builds produced?

Comment (Eilon) [reply Documentation/ArcadeCommunications.md]: Some of that will be covered by @shawnro 's team broadly in these issues:

- Service KPI’s – https://github.com/dotnet/core-eng/issues/5509
- Product KPI’s - https://github.com/dotnet/arcade/issues/2629

File: Documentation/ArcadeCommunications.md (hunk 1 above)
Comment (Chrisboh) [reply Documentation/ArcadeCommunications.md]: +1 to what Matt said.  The Python 3 change is a perfect example of a type of change we make that we would need this ability to have.

File: Documentation/ArcadeCommunications.md (hunk 2 above)
Comment (shawnro) [reply Documentation/ArcadeCommunications.md]: Guidance and best practices comms is definitely something we'll want to highlight.  I'd expect it not to be super frequent, but worth calling out.

File: Documentation/ArcadeCommunications.md (hunk 1 above)
Comment (shawnro) [reply Documentation/ArcadeCommunications.md]: I don't disagree, but this is something that needs to be exceptional.  I'd like to understand why the Python 3 change would fall into this category.

File: Documentation/ArcadeCommunications.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+- **Disruptive** - Change doesn’t break, but either adds debt with the product teams OR is considered risky
+  - Examples: release pipelines, licensing, cert removal
+- **Minimal risk** - Some risk, but likely will be fine (these might also be called “updates of note”)
+  - Examples: new feature, notable bug fix, etc. 
+- **Minor, low risk** - Very low risk, and most folks won't care.  
+  - Most changes are these last two bullets.
+  - Examples: security update, minor bug fixes, etc
+
+## Engineering Services Policy for Updates
+- Product release schedules are published and kept "top of mind" for the team
```
Comment (shawnro) [inline Documentation/ArcadeCommunications.md]: Can you clarify, is this .NET Core product or our services as a product?  Ideally we'll have a published (or predictable) schedule for updates to our services as a product.

File: Documentation/ArcadeCommunications.md (hunk 4 above)
Comment (shawnro) [reply Documentation/ArcadeCommunications.md]: This is a good question thought - @Chrisboh, @markwilkie should we include communication of KPI's in this doc?

File: Documentation/ArcadeCommunications.md [hunk 6]
```diff
@@ -0,0 +1,40 @@
+# Types of Arcade Communication
```
Comment (shawnro) [inline Documentation/ArcadeCommunications.md]: Would love to see us highlight the goals and/or what successful communication here looks like.

File: Documentation/ArcadeCommunications.md (hunk 1 above)
Comment (markwilkie) [reply Documentation/ArcadeCommunications.md]: done

File: Documentation/ArcadeCommunications.md (hunk 4 above)
Comment (markwilkie) [reply Documentation/ArcadeCommunications.md]: I was assuming (we all know what that does...grin) that the KPI's are reports which are always available.  Right now, this doc is more about how we update folks when things change.  However...thoughts?

File: Documentation/ArcadeCommunications.md (hunk 5 above)
Comment (markwilkie) [reply Documentation/ArcadeCommunications.md]: You mean, as in "product release schedules"??  grin.  I'll also add .NET Core to the front to minimize confusion.

File: Documentation/ArcadeCommunications.md (hunk 6 above)
Comment (markwilkie) [reply Documentation/ArcadeCommunications.md]: Added a whole section

File: Documentation/ArcadeCommunications.md (hunk 4 above)
Comment (shawnro) [reply Documentation/ArcadeCommunications.md]: I'd like us to do some sort of regular "push" of the KPI's once we have them in place.  However, whether or not that's part of this, doesn't matter as much to me.

File: Documentation/ArcadeCommunications.md [hunk 7]
```diff
@@ -0,0 +1,48 @@
+# Types of Engineering Services Change Communication
+
+## Goals and Success Criteria
+The primary goal is to communicate updates, new features, and anything of note that is changing.  Success looks like:
+- No surprises
+- New value is known
+- Breaking changes are expected, and proper runway is given to mitigate
+- All changes (big or small) accessible for investigations
```
Comment (shawnro) [inline Documentation/ArcadeCommunications.md]: I like it.  What does "accessible for investigations" mean?

Comment (markwilkie) [reply Documentation/ArcadeCommunications.md]: I was trying to say that all changes to all aspect of our infra are available to devs trying to debug or understand what changed under them - which is usually the main reason for these exhaustive change logs.  I'll try and think of better language....

**Comment by markwilkie** [comment]
Added 'breaking change' doc

File: Documentation/ArcadeCommunications.md (hunk 7 above)
Comment (shawnro) [reply Documentation/ArcadeCommunications.md]: Transparency that enables easy diagnosability?

Comment (markwilkie) [reply Documentation/ArcadeCommunications.md]: done

**Comment by ilyas1974** [comment]
This process seems a bit confusing to me. Is this to provide guidance for "emergency"/one-off changes that may have impact to customers or "regularly scheduled" updates (or do regularly scheduled updates not exist yet)? How are major versions defined? 

Also, many of the links in the ChangePolicy.md document give me a 404 error. I don't know if this is a permissions issues or a broken link.

**Comment by markwilkie** [comment]
> This process seems a bit confusing to me. Is this to provide guidance for "emergency"/one-off changes that may have impact to customers or "regularly scheduled" updates (or do regularly scheduled updates not exist yet)?

Any breaking change falls under this - either emergency or regularly scheduled.

>How are major versions defined?

As defined by SemVer  (https://semver.org/)  Also, I called out semver explicitly in the doc now.

>Also, many of the links in the ChangePolicy.md document give me a 404 error. I don't know if this is a permissions issues or a broken link.

I'll double check the links

**Comment by markwilkie** [comment]
I'll plan to merge this tomorrow or Friday unless I get any pushback.

File: Documentation/Policy/ArcadeCommunications.md [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+
+## Goals and Success Criteria
+The primary goal is to communicate updates, new features, and anything of note that is changing.  Success looks like:
+- No surprises
+- New value is known
+- Breaking changes are expected, and proper runway is given to mitigate
+- All changes (big or small) available to help with transparency that enables easy diagnosability
+
+## Areas to Communicate Changes 
+There are several areas around which  communication will occur.  The nature of these updates remain the same for each of these areas.
```
Comment (riarenas) [inline Documentation/Policy/ArcadeCommunications.md]:
```suggestion
There are several areas around which communication will occur.  The nature of these updates remain the same for each of these areas.
```

File: Documentation/Policy/ArcadeCommunications.md [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+- DNC Services (Helix, Darc, etc)
```
Comment (riarenas) [inline Documentation/Policy/ArcadeCommunications.md]: Should this be DNCeng?

```suggestion
- DNCeng Services (Helix, Darc, etc) 
```

File: Documentation/Policy/ArcadeCommunications.md [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+- No updates to shared services during stabilization period (often a week) leading up to release
+- Published bar for when we’ll take changes [Arcade Change Bar](ChangeBar.md) 
+- Published bar used for determining class of update/change [Arcade Change Bar](ChangeBar.md) 
+
+## Communication Mechanics/Methods
+- Breaking changes follow procedure [Arcade Changes Policy document](ChangesPolicy.md) 
+- Disruptive changes follow procedure [Arcade Changes Policy document](ChangesPolicy.md) 
+- Release notes with “updates of note” (human crafted)
+  - Done every time an update is deployed for our services
+  - Done at most weekly for tools (e.g. Arcade SDK).
```
Comment (riarenas) [inline Documentation/Policy/ArcadeCommunications.md]: We have multiple contributors to Arcade from different teams. What's the expectation for how the release notes for those are handled?

Comment (riarenas) [inline Documentation/Policy/ArcadeCommunications.md]:
```suggestion
  - Done at most weekly for tools (e.g. Arcade SDK)
```

File: Documentation/Policy/ChangesPolicy.md [hunk 11]
```diff
@@ -0,0 +1,36 @@
+# Arcade Breaking Changes Policy
+
+## Overview
+In a nutshell, no breaking changes within a major version (SemVer rules) except where absolutely necessary.  To be sure there are some nuances to this, but in general - Arcade doesn't do breaking changes.  
+
+## Validation:
+In order to no break Arcade consumers, there must be comprehensive validation *before* pushing to the latest channel.  From there, it's expected that the repo's PR validation would catch things specific to that repo.  More specific here. [Arcade Validation Document](../Validation/Overview.md)
```
Comment (riarenas) [inline Documentation/Policy/ChangesPolicy.md]:
```suggestion
In order to not break Arcade consumers, there must be comprehensive validation *before* pushing to the `latest` channel.  From there, it's expected that the repo's PR validation would catch things specific to that repo.  More specific here. [Arcade Validation Document](../Validation/Overview.md)
```

File: Documentation/Policy/ChangesPolicy.md [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+1. Two days later, a 'we're doing this now' email to ‘dncpartners’
+1. Proactively provide support and/or help as required 
+
+# Arcade Disruptive Changes Policy
+
+## Overview
+From time to time, it is necessary or desirable to make a change which is not technically breaking, but is likely disruptive. The process is largely the same as breaking changes.
+
+## Validation:
+In order to no break Arcade consumers, there must be comprehensive validation *before* pushing to the latest channel.  From there, it's expected that the repo's PR validation would catch things specific to that repo.  More specific here. [Arcade Validation Document](../Validation/Overview.md)
```
Comment (riarenas) [inline Documentation/Policy/ChangesPolicy.md]: [nit] 
```suggestion
In order to not break Arcade consumers, there must be comprehensive validation *before* pushing to the latest channel.  From there, it's expected that the repo's PR validation would catch things specific to that repo.  More specific here. [Arcade Validation Document](../Validation/Overview.md)
```

File: Documentation/Policy/ArcadeCommunications.md (hunk 10 above)
Comment (markwilkie) [reply Documentation/Policy/ArcadeCommunications.md]: My intent was to describe the expectation in the 'Goals and Success Criteria' section at the top.  Are there still areas that you'd like to see clarified further?

Comment (riarenas) [reply Documentation/Policy/ArcadeCommunications.md]: My doubt is mostly on the mechanics for this particular section.
We're going to manually craft release notes for shared tools. Who is going to craft the notes? the eng services team, or should we encourage the individuals that contribute features to Arcade to help out?

If that's too specific for the scope of this document, that's also fine.

Comment (markwilkie) [reply Documentation/Policy/ArcadeCommunications.md]: Ah I see - yea, I was staying away from the mechanics at least in this doc.  But yes, that'll have to be worked out for sure.

**Pull request merged**